### PR TITLE
DBZ-2480 Initial commit of outbox event router SMT doc updated for 1.2

### PR DESCRIPTION
**Pull request by TovaCohen**
This is for [DBZ-2480](https://issues.redhat.com/browse/DBZ-2480).
I revised the doc for the outbox event router SMT. The downstream Technology Preview note is conditionalized to show downstream.  Comments contain annotations that allow the downstream fetch script to split this file into the modular files required downstream. 
Let me know what needs to change. 
When this is merged, please backport this to 1.2.
Thanks!

### CODE REVIEW

**Review by Naros — COMMENTED** [review]
The only concern I have is that we refer to the Outbox EventRouter in multiple places like `outbox.EventRouter` and I don't particularly like that, but if others are fine with that, I'm willing to dissent and go with the majority.

@gunnarmorling @jpechane what is your take here?

**Comment by TovaCohen** [comment]
@Naros Just want to note that referring to `outbox.EventRouter` is consistent with referring to `ByLogicalTableRouter` and `ExtractNewRecordState`. I included the `outbox` part because `EventRouter` seemed to general. 

I'm happy to do whatever you all want. But I hope we can use a consistent convent to refer to the SMTs by name.

File: documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc [hunk 1]
```diff
@@ -9,26 +13,56 @@
...
 ====
+endif::product[]
+
+See link:https://debezium.io/blog/2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern/[Reliable Microservices Data Exchange With the Outbox Pattern] to learn about why the outbox pattern is useful and how it works. 
 
-The Outbox Event Router SMT has the intent to provide out-of-the-box support for the Outbox Pattern.
-For more information on why and how the Outbox Pattern is used please refer our blog post link:/blog/2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern/[Reliable Microservices Data Exchange With the Outbox Pattern].
+ifdef::community[]
+For details about how {prodname} implements the outbox pattern, see {jira-url}/browse/DBZ-1169[Outbox SMT Design]. For an example that you can run, see the  link:https://github.com/debezium/debezium-examples/tree/master/outbox[outbox pattern demo], which is in the {prodname} examples repository. It includes an example of how to configure a {prodname} connector to run the `outbox.EventRouter` SMT.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc:40]: Let's remove the reference to the Jira issue, it's not really needed in the docs.

**Review by gunnarmorling — COMMENTED** [review]
@TovaCohen, could you publish a rendered vesion of this doc? That'd help to see what the downstream version of it will look like. Thanks!

**Comment by Naros** [comment]
@TovaCohen I discussed this with @gunnarmorling and we both agree that it makes more sense from a docs perspective that unless we're referring to a place where you configure the actual outbox event router class name in the configuration, it would be better to have references of `outbox.EventRouter` changed simply to `outbox event router`.  I understand the desire for consistency with the other router docs, but I now wonder if we shouldn't have the others conform to this style instead?

File: documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc [hunk 2]
```diff
@@ -9,26 +13,56 @@
 
 toc::[]
 
+The outbox pattern is a way to ensure that related services that need the same data updates get those updates in a consistent manner. To implement the outbox pattern in a {prodname} application, configure a {prodname} connector to:
```
[the remark this answers is not in the document]
Comment (TovaCohen) [reply documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc:16]: @gunnarmorling Is this correct:
The outbox pattern is a way to safely and reliably exchange data between multiple (micro) services. An outbox pattern implementation avoids inconsistencies between a service's internal state (as typically persisted in its database) and state in events consumed by services that need the same data.

File: documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc (hunk 1 above)
Comment (TovaCohen) [reply documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc:40]: Done in the next commit.

**Comment by TovaCohen** [comment]
Why do you think `outbox event router` is better than `outbox.EventRouter`?  And just to be clear and make sure we are talking about the same thing - both those instances are in fixed-width, monospace, font.  Where is `outbox event router` a literal that should be in fixed-width?

I'm not trying to be difficult, but I can do a better job with the doc in the first place when I understand why things should be a certain way.

**Comment by Naros** [comment]
> Where is outbox event router a literal that should be in fixed-width?

What I meant was rather than using the fixed-width, monospace font it should read "outbox event router" in normal font of the paragraph for which its contained.

File: documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (TovaCohen) [reply documentation/modules/ROOT/pages/configuration/outbox-event-router.adoc]: I updated the doc with my suggested paragraph. It is in the commit I just added to this PR. Let me know if that needs to change.

**Comment by TovaCohen** [comment]
Oh, got it.  Normal font. I'll make those changes now.

**Comment by TovaCohen** [comment]
@gunnarmorling @Naros  Here is a link to the downstream user guide, outbox router section:
https://access.redhat.com/documentation/en-us/red_hat_integration/2020-q3/html-single/debezium_user_guide/index?lb_target=stage#outbox-event-router
You must be logged in to the customer portal for the link to work.

**Review by Naros — APPROVED** [review]

**Comment by Naros** [comment]
Applied to master & branch 1.2.  Thanks @TovaCohen!

**Pull request merged**